### ISSUE TITLE
Incorrect Libs.private entry for libcrypto.pc

### ISSUE THREAD

**Issue by jkolo** [openssl/openssl#24986]
After updating openssl to version 3.3.1, the pkg-config for libcrypto is incorrectly returned. Example output:
```
❯ pkg-config openssl --libs --static
-lssl -lcrypto -lz -ldl -pthread -Wl --push-state --as-needed -latomic --pop-state
```
correct output from version 3.2.2:
```
❯ pkg-config openssl --libs --static
-lssl -lcrypto -lz -ldl -pthread -Wl,--push-state,--as-needed,-latomic,--pop-state
```
The problem is in the file `/usr/lib/pkgconfig/libcrypto.pc`, which in version 3.3.1 has the following content:
```
libdir=/usr/lib
includedir=/usr/include
enginesdir=/usr/lib/engines-3
modulesdir=/usr/lib/ossl-modules

Name: OpenSSL-libcrypto
Description: OpenSSL cryptography library
Version: 3.3.1
Libs: -L${libdir} -lcrypto
Libs.private: -lz -ldl -pthread -Wl --push-state --as-needed -latomic --pop-state
Cflags: -I${includedir}
```
And in version 3.2.2 it looked like this:
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include
enginesdir=${libdir}/engines-3
modulesdir=${libdir}/ossl-modules

Name: OpenSSL-libcrypto
Description: OpenSSL cryptography library
Version: 3.2.2
Libs: -L${libdir} -lcrypto
Libs.private: -lz -ldl -pthread -Wl,--push-state,--as-needed,-latomic,--pop-state
Cflags: -I${includedir}
```

**Comment by nhorman** [comment]
this is likely fixed by https://github.com/openssl/openssl/pull/24687

It missed 3.3.1, but will be available in 3.3.2.  If you build from the head of the master or openssl-3.3 branches, you should pick it up

**Comment by floppym** [comment]
I don't think this is fixed by https://github.com/openssl/openssl/pull/24687.

The issue is that LDLIBS gets split on both whitespace and commas in util/mkinstallvars.pl. In this case we do NOT want to split on commas.

The input argument was `-Wl,--push-state,--as-needed,-latomic,--pop-state`, and this should be treated as a single argument when passed to the compiler.

Downstream bug report: https://bugs.gentoo.org/936793

**Comment by levitte** [comment]
It's not fixed by #24687, indeed.  Quick PR coming up, this is a very easy fix.

**Comment by satmandu** [comment]
This regression in 3.3.1 still appears to be a problem on the 3.3 branch.

Having just built from it for Linux/Chromebrew, I am seeing `libdir` borked in all of the pkgconfig files:
```
cat /usr/local/lib/pkgconfig/libcrypto.pc
prefix=/usr/local
exec_prefix=${prefix}
libdir=${exec_prefix}/.
includedir=${prefix}/include
enginesdir=${libdir}/lib/engines-3
modulesdir=${libdir}/lib/ossl-modules

Name: OpenSSL-libcrypto
Description: OpenSSL cryptography library
Version: 3.3.2-dev
Libs: -L${libdir} -lcrypto
Libs.private: -ldl -pthread
Cflags: -I${includedir}
```

```
cat /usr/local/lib/pkgconfig/libssl.pc
prefix=/usr/local
exec_prefix=${prefix}
libdir=${exec_prefix}/.
includedir=${prefix}/include

Name: OpenSSL-libssl
Description: Secure Sockets Layer and cryptography libraries
Version: 3.3.2-dev
Requires.private: libcrypto
Libs: -L${libdir} -lssl
Cflags: -I${includedir}
```

```
cat /usr/local/lib/pkgconfig/openssl.pc
prefix=/usr/local
exec_prefix=${prefix}
libdir=${exec_prefix}/.
includedir=${prefix}/include

Name: OpenSSL
Description: Secure Sockets Layer and cryptography libraries and tools
Version: 3.3.2-dev
Requires: libssl libcrypto
```

**Comment by levitte** [comment]
@satmandu, that's a different issue.  Do you mind raising it as a new issue?
